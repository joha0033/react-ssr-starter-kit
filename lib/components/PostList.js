import React from 'react'

import Post from './Post'

const PostList = ( props ) => {
  return (
    <div>
      {
        Object.values(props.posts).map((post, index) => {
          return (
            <Post 
              key={index}
              post={post}
              actions={props.postActions}
            />
          )
        })            
      }
    </div>
  )
}

export default PostList