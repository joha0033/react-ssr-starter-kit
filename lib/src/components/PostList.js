import React from 'react'
import Post from './Post'
import { postListActions } from '../_actions/postList.actions'
import { connect } from 'react-redux'

class PostList extends React.Component {
  componentDidMount() {
    this.props.fetchPosts()
  }
  
  render() {
    const posts = () => {
      let arr = Object
        .values(this.props.postList.posts)
        .map((post, index) => {
          let { firstName, lastName } = this.props.postList.users[post.user_id]
          let author = firstName + ' ' + lastName
            
          let postWithAuthor = {
            ...post,
            author
          }
          return (
            <Post 
              key={index}
              post={postWithAuthor}
            />
          )
        }) 
      return arr
    }

    const loading = () => (<div>LOADING... </div>)

    return ( 
      <div>
        {
          this.props.postList.loading
            ? loading()
            : posts()           
        }
      </div>
    )
  }
}
 
const mapStateToProps = (state) => {
  const { postList } = state
  return {
    postList
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchPosts: () => {
      dispatch(postListActions.fetchPosts())
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostList)