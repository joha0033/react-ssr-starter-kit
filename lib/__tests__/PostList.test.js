import React from 'react'
import PostList from '../components/PostList'

import renderer from 'react-test-renderer'

describe('PostList', () => {
  it('renders post list', () => {

    const testProps = {
      posts: {
        a: { id: 'a' },
        b: { id: 'b' }
      },
      store:{
        lookupAuthor: jest.fn(() => ({}))
      } 
    }

    const tree = renderer.create(
      <PostList 
        { ...testProps}
      />
    ).toJSON()
    
    expect(tree.children.length).toBe(2)
    expect(tree).toMatchSnapshot()
  })
})