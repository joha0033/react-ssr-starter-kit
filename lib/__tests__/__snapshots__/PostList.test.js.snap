// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders post list 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#abc",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 2,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Author: 
       
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
          "paddingRight": 10,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#abc",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 2,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Author: 
       
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
          "paddingRight": 10,
        }
      }
    />
  </div>
</div>
`;
